src/main/kotlin/net/irisshaders/lilybot/extensions/events/LogUploading.kt

```diff
@@ -167,23 +172,27 @@ class LogUploading : Extension() {
 			}
 		}
 	}
+	@kotlinx.serialization.Serializable
+	data class logClass(val success: Boolean, val id: String, val url: String, val raw: String)
 
 	private suspend fun postToHasteBin(text: String): String {
 		val client = HttpClient()
 
-		var response = client.post<HttpResponse>("https://www.toptal.com/developers/hastebin/documents") {
-			body = text
+		val response = client.post<HttpResponse>("https://api.mclo.gs/1/log") {
+			body = FormDataContent(Parameters.build {
+				append("content", text)
+			})
 		}.content.toByteArray().decodeToString()
-
-		if (response.contains("\"key\"")) {
-			response = "https://www.toptal.com/developers/hastebin/" + response.substring(
-				response.indexOf(":") + 2,
-				response.length - 2
-			)
+		val log = Json.decodeFromString<logClass>(response)
+		if (log.success == false){
+			throw Exception("Failed to upload log")
+			/* mclo.gs returns the error but I can't think
```

Comment:
There must be someway around this, shall have to look into it

Reply:
after some further thinking, i believe that if i have the error property be optional it could fix things